src/activity_handlers.cpp

```diff
@@ -2637,28 +2639,27 @@ void activity_handlers::craft_do_turn( player_activity *act, player *p )
         p->cancel_activity();
         return;
     }
-    act->moves_left -= crafting_speed * p->get_moves();
-    p->set_moves( 0 );
     if( calendar::once_every( 1_hours ) && crafting_speed < 0.75f ) {
         // TODO: Describe the causes of slowdown
         p->add_msg_if_player( m_bad, _( "You can't focus and are working slowly." ) );
     }
-}
 
-void activity_handlers::craft_finish( player_activity *act, player *p )
-{
-    p->complete_craft();
+    craft.item_counter += crafting_speed * p->get_moves();
+    p->set_moves( 0 );
     act->set_to_null();
-}
 
-void activity_handlers::longcraft_finish( player_activity *act, player *p )
-{
-    const int batch_size = act->values.front();
-    p->complete_craft();
-    act->set_to_null();
-    // Workaround for a bug where longcraft can be unset in complete_craft().
-    if( p->making_would_work( p->lastrecipe, batch_size ) ) {
-        p->last_craft->execute();
+    if( craft.item_counter >= rec.time ) {
+        item craft_copy = craft;
+        p->i_rem( pos );
```

Comment:
You don't need to make a copy yourself: `i_rem` makes it for you (even if you never asked for this).

Reply:
So it does, thanks!